user/views.py

```diff
@@ -1,3 +1,444 @@
-from django.shortcuts import render
+import json, re, bcrypt, jwt
 
-# Create your views here.
+from django.views   import View
+from django.http    import JsonResponse
+
+from my_settings    import SECRET, ALGORITHM
+from user.models    import Review, User, Gender, ShoppingBasket, FrequentlyPurchasedProduct, UserRank
+from product.models import Product, ProductOption
+from core.utils     import access_decorator
+
+class SignUp(View): # 회원가입
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            for key in data.keys():
+                if data['user_id'] == '' or data['password'] == '' or data['user_name'] == '' or data['phone'] == '' or data['address'] == '':
+                    return JsonResponse({'message' : 'NOT_ENTERED_KEY'}, status = 400)
+            
+            password = data['password']
+            hashed_pw = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())
+
+            User.objects.create(
+                user_id           = data['user_id'],
+                password          = hashed_pw.decode('utf-8'),
+                user_name         = data['user_name'],
+                email             = data['email'],
+                phone             = data['phone'],
+                address           = data['address'],
+                gender            = Gender(id = data['gender']),
+                date_of_birth     = data['date_of_birth'],
+                recommender       = data['recommender'],
+                event             = data['event'],
+                is_privacy_policy = data['is_privacy_policy'],
+                is_sms_agreed     = data['is_sms_agreed'],
+                is_email_agreed   = data['is_email_agreed'],
+            )
+            
+            return JsonResponse({'message' : 'SUCCESS'}, status = 200)
+
+        except KeyError as ex:
+            return JsonResponse({'message' : 'KEY_ERROR_' + ex.args[0]}, status = 400)
+        except Exception as ex:
+            return JsonResponse({'message' : 'ERROR_' + ex.args[0]}, status = 400)
+
+class CheckID(View): # 아이디 중복확인
```

Comment:
```suggestion
class CheckIDView(View): # 아이디 중복확인
```
클래스 이름에 `View` 를 붙여서 통일해주세요 :)

Reply:
넵 수정했습니다 :)